diff --git a/td/example/ios/build-openssl.sh b/td/example/ios/build-openssl.sh
--- a/td/example/ios/build-openssl.sh
+++ b/td/example/ios/build-openssl.sh
@@ -8,7 +8,7 @@ git reset --hard || exit 1
 git apply ../Python-Apple-support.patch || exit 1
 cd ..
 
-platforms="macOS iOS watchOS tvOS visionOS"
+platforms="macOS"
 
 for platform in $platforms;
 do
@@ -25,7 +25,7 @@ do
     echo $platform
     cd Python-Apple-support
     #NB: -j will fail
-    SOURCE_DATE_EPOCH=1 ZERO_AR_DATE=1 make OpenSSL-$platform || exit 1
+    SOURCE_DATE_EPOCH=1 ZERO_AR_DATE=1 make OpenSSL-$platform TARGETS-macOS=macosx.arm64 || exit 1
     cd ..
     rm -rf third_party/openssl/$platform || exit 1
     mkdir -p third_party/openssl/$platform/lib || exit 1
diff --git a/td/example/ios/build.sh b/td/example/ios/build.sh
--- a/td/example/ios/build.sh
+++ b/td/example/ios/build.sh
@@ -21,7 +21,7 @@ set_cmake_options () {
   options="$options -DCMAKE_BUILD_TYPE=Release"
 }
 
-platforms="macOS iOS watchOS tvOS visionOS"
+platforms="macOS"
 #platforms="watchOS"
 for platform in $platforms;
 do
@@ -35,7 +35,7 @@ do
   for simulator in $simulators;
   do
     if [[ $platform = "macOS" ]]; then
-      other_options="-DCMAKE_OSX_ARCHITECTURES='x86_64;arm64'"
+      other_options="-DCMAKE_OSX_ARCHITECTURES=arm64"
     else
       if [[ $platform = "watchOS" ]]; then
         ios_platform="WATCH"
